### PR TITLE
Various fixes/improvements

### PR DESCRIPTION
**Pull request by dpfrey**
See the commit messages for detail.  I don't think any of these commits will introduce changes in behavior.

### CODE REVIEW

File: location/location.c [hunk 1]
```diff
@@ -78,21 +91,17 @@ void getLocation(le_timer_Ref_t timerRef) {
...
- * Note that we run this on a timer to avoid
- * blocking up the main (only) thread. If this
- * was run in a while(true) that sleeps,
- * the IPC caller would be blocked indefinitely
+ * Note that we get the position from a timer to avoid blocking up the main
+ * (only) thread. If this was run in a while(true) that sleeps, the IPC caller
+ * would be blocked indefinitely.
  */
-le_result_t gps_init() {
+COMPONENT_INIT {
```
Comment (nvandoorn) [inline location/location.c]: I personally liked having this in a sub-routine since we may have a few init sub-routines down the road.

**Review by nvandoorn — CHANGES_REQUESTED** [review]
I like all of this except factoring everything `COMPONENT_INIT`. Revert that and I'll merge this.

**Comment by nvandoorn** [comment]
I should also make the requirement of our [util](https://github.com/brnkl/legato-util) a little clearer. I'll open an issue for this and solve it in a subsequent PR.

File: location/location.c (hunk 1 above)
Comment (dpfrey) [reply location/location.c]: done

**Review by nvandoorn — APPROVED** [review]
Looks great! Thanks for the contribution @dpfrey

**Pull request merged**